packages/flutter_tools/test/asset_bundle_package_fonts_test.dart

```diff
@@ -0,0 +1,293 @@
+
+      writePubspecFile('pubspec.yaml', 'test');
+      writePackagesFile('test_package:p/p/lib/\ntest_package2:p2/p/lib/');
+      final String fontsSection = '''
+       - family: foo
+         fonts:
+           - asset: packages/test_package2/bar
+''';
+      writePubspecFile(
+        'p/p/pubspec.yaml',
+        'test_package',
+        fontsSection: fontsSection,
+      );
+      writePubspecFile('p2/p/pubspec.yaml', 'test_package2');
+
+      final String font = 'bar';
+      writeFontAsset('p2/p/lib/', font);
+
+      final String expectedFontManifest =
+          '[{"fonts":[{"asset":"packages/test_package2/bar"}],'
+          '"family":"packages/test_package/foo"}]';
+      await buildAndVerifyFonts(
+        <String>[],
+        <String>[font],
+        <String>['test_package2'],
+        expectedFontManifest,
+      );
+    }, overrides: contextOverrides);
+
+    testUsingContext('App uses package font with properties ans own font file', () async {
+      establishFlutterRoot();
+
+      writePubspecFile('pubspec.yaml', 'test');
+      writePackagesFile('test_package:p/p/lib/');
+
+      final String pubspec = '''
+       - family: foo
+         fonts:
+           - style: italic
+             weight: 400
+             asset: a/bar
+''';
+      writePubspecFile(
+        'p/p/pubspec.yaml',
+        'test_package',
+        fontsSection: pubspec,
+      );
+      final String font = 'a/bar';
+      writeFontAsset('p/p/', font);
+
+      final String expectedFontManifest =
+          '[{"fonts":[{"weight":400,"style":"italic","asset":"packages/test_package/a/bar"}],'
+          '"family":"packages/test_package/foo"}]';
+      await buildAndVerifyFonts(
+        <String>[],
+        <String>[font],
+        <String>['test_package'],
+        expectedFontManifest,
+      );
+    }, overrides: contextOverrides);
+
+    testUsingContext('App uses local font and package font with own font file.', () async {
+      establishFlutterRoot();
+
+      final String fontsSection = '''
+       - family: foo
+         fonts:
+           - asset: a/bar
+''';
+      writePubspecFile(
+        'pubspec.yaml',
+        'test',
+        fontsSection: fontsSection,
+      );
+      writePackagesFile('test_package:p/p/lib/');
+      writePubspecFile(
+        'p/p/pubspec.yaml',
+        'test_package',
+        fontsSection: fontsSection,
+      );
+
+      final String font = 'a/bar';
+      writeFontAsset('', font);
+      writeFontAsset('p/p/', font);
+
+      final String expectedFontManifest =
+          '[{"fonts":[{"asset":"packages/test_package/a/bar"}],'
+          '"family":"packages/test_package/foo"},'
+          '{"fonts":[{"asset":"a/bar"}],"family":"foo"}]';
+      await buildAndVerifyFonts(
+        <String>[font],
+        <String>[font],
+        <String>['test_package'],
+        expectedFontManifest,
+      );
+    }, overrides: contextOverrides);
+  });
+}
+
+Map<Type, Generator> get contextOverrides =>
```

Comment:
Can't this be on one line? We should not use `=>` with multi-line syntax.

Reply:
It's quite long, changed it to use`{}`.